packages/extension-api/src/extension-api.d.ts

```diff
@@ -1507,6 +1507,27 @@ declare module '@podman-desktop/api' {
      */
     export function showErrorMessage(message: string, ...items: string[]): Promise<string | undefined>;
 
+    /**
+     * Create a Task inside the TaskManager with a progress bar, by default indeterminate for the time of the promise you gave.
+     * If the promise complete without raising an exception the task will be marked as completed, otherwise it will be marked as failed.
+     *
+     * ```ts
+     * try {
+     *  void podmanDesktopApi.window.withProgress<void>(
+     *       { location: podmanDesktopApi.ProgressLocation.TASK_WIDGET, title: `Running task` },
+     *       async progress => {
+     *           // might throw an error
+     *           await complicatedTask();
+     *       },
+     *     );
+     * } catch (error) {
+     *    // to something with the error
+     * }
+     * ```
```

Comment:
I would have the version without error first and another example with catch


```suggestion
     * ```ts
     * await window.withProgress(
     *       { location: ProgressLocation.TASK_WIDGET, title: `Running task` },
     *       async progress => {
     *          progress.report({message: 'task1' });
     *          await task1();
     *          progress.report({increment: 50, message: 'task2' });
     *          await task2();
     *       },
     *     );
     * ```
    * // in case of error during the callback, the error is rethrown so you'll need to use try/catch around the call to withProgress
     * ```ts
     * try {
     *   await window.withProgress(
     *       { location: ProgressLocation.TASK_WIDGET, title: `Running task` },
     *       async progress => {
     *          throw new Error('error when running a task');
     *       },
     *     );
     * } catch (error) {
     *    // do something with the error
     * }
     * ```
     
     
     
     
```